lib/pkgcloud/google/storage/client/files.js

```diff
@@ -41,36 +40,26 @@ exports.upload = function (options) {
   var self = this,
     bucket = this._getBucket(options),
     file = this._getFile(bucket, options),
-    uploadOptions = {};
+    metadata = { contentType: options.contentType };
 
   // check for deprecated calling with a callback
   if (typeof arguments[arguments.length - 1] === 'function') {
     self.emit('log::warn', 'storage.upload no longer supports calling with a callback');
   }
 
-  if (options.contentType) {
-    uploadOptions.contentType = options.contentType;
-  } else {
-    uploadOptions.contentType = mime.getType(options.file || options.remote || options);
-  }
-
   var proxyStream = through(),
-    writableStream = file.createWriteStream(uploadOptions);
+    writableStream = file.createWriteStream({ metadata });
```

Comment:
A comment here about implicit mime-type detection will help future maintainers / editors of this code.

Reply:
done in https://github.com/pkgcloud/pkgcloud/pull/666/commits/32eccf5c04738bbe653e84054ee2f7485baff99a